toolkit/tools/pkg/imagecustomizerlib/resolvconf.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+)
+
+type resolvConfType int
+
+const (
+	resolvConfTypeNone resolvConfType = iota
+	resolvConfTypeSymlink
+	resolvConfTypeFile
+)
+
+type resolvConfInfo struct {
+	existingType resolvConfType
+	fileContents string
+	symlinkPath  string
+}
+
+const (
+	resolvConfPath        = "/etc/resolv.conf"
+	resolvSystemdStubPath = "../run/systemd/resolve/stub-resolv.conf"
```

Comment:
Is there a reason to not use an absolute path for `resolvSystemdStubPath`? I doubt that `resolvConfPath` will ever change but if not needed, I think it's unnecessary to bind `resolvSystemdStubPath` to `resolvConfPath`. Or at least I think these two paths are independent of each other as in: regardless of what `resolvConfPath` may be, `resolvSystemdStubPath` may change independently.

Reply:
This is the value that `systemd-resolved` uses when it sets up the symlink on first boot. I am just copying it.